src/Microsoft.DotNet.Darc/src/Darc/Options/SubscriptionsCommandLineOptions.cs

```diff
@@ -56,7 +56,7 @@ abstract class SubscriptionsCommandLineOptions : CommandLineOptions
     [Option("ids", Separator = ',', HelpText = "Get only subscriptions with these ids.")]
     public IEnumerable<string> SubscriptionIds { get; set; }
 
-    public async Task<IEnumerable<Subscription>> FilterSubscriptions(IRemote remote)
+    public virtual async Task<IEnumerable<Subscription>> FilterSubscriptions(IRemote remote)
```

Comment:
Is this made virtual just for the sake of the tests?

Wouldn't it be better to mock `IRemote`?

Reply:
I think it would be better to mock it

Reply:
@RussKie you could add an internal ctor that accepts the DI container (it's available on the parent already) and you can inject your own services in that way.

Reply:
> I think it would be better to mock it

You can't mock non-virtual members:
```
Message: 
System.NotSupportedException : Invalid setup on a non-virtual (overridable in VB) member: t => t.FilterSubscriptions(It.IsAny<IRemote>())
```


> you could add an internal ctor that accepts the DI container (it's available on the parent already) and you can inject your own services in that way.

I entertained that possibility, but I didn't see any way it could help in this specific situation. Perhaps I miss the way you think it may work, and if you give me some hints, I'm happy to look again.

The core issue in my view is this line:
https://github.com/dotnet/arcade-services/blob/0e030ad4e533162d3bc53c9846770b368f10981d/src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionsOperation.cs#L34
...or more specifically this `static` method:
https://github.com/dotnet/arcade-services/blob/0e030ad4e533162d3bc53c9846770b368f10981d/src/Microsoft.DotNet.Darc/src/Darc/Helpers/RemoteFactory.cs#L81-L84

Because it is static, it can't be mocked, this means I have no control over the provided `IRemote` instance, and as a result `FilterSubscriptions()` will do an unknown amount of work, which isn't acceptable in the test environment.


One way out of this could be turning `GetBarOnlyRemote` into an instance method and injecting `RemoteFactory` via the DI. However, this work is completely out of the scope of this FR.

Reply:
@RussKie what I meant is that there is this ctor on `Operation`:
https://github.com/dotnet/arcade-services/blob/0e030ad4e533162d3bc53c9846770b368f10981d/src/Microsoft.DotNet.Darc/src/Darc/Operations/Operation.cs#L22

You could register the `IRemote` either by passing it to an internal ctor or, when no `IRemote`, pass the static factory method.

Then, you can use the `Provider` member for resolving the `IRemote` service instead of this:
https://github.com/dotnet/arcade-services/blob/0e030ad4e533162d3bc53c9846770b368f10981d/src/Microsoft.DotNet.Darc/src/Darc/Operations/GetSubscriptionsOperation.cs#L34

Reply:
Updated, please have a look.